csharp-design-patterns/UnitOfWork/UnitOfWork/DataAccess/EntityFramework/AppDbContext.cs

```diff
@@ -0,0 +1,22 @@
+using Microsoft.EntityFrameworkCore;
+using UnitOfWork.Entities;
+
+namespace UnitOfWork.DataAccess.EntityFramework;
+
+public class AppDbContext : DbContext, IDatabase, IUnitOfWork
+{
+    public async Task<ITransaction> BeginTransactionAsync()
+    {
+        var transaction = await Database.BeginTransactionAsync();
+        return new EfTransaction(transaction);
+    }
+    
+    public DbSet<Order> Orders { get; set; }
```

Comment:
In a class, properties should always be located above the public methods.

Reply:
reordered them, in `DummyDatabase` also